tools/test_full_sync.py

```diff
@@ -51,25 +88,50 @@ async def run_sync_test(file: Path, db_version=2) -> None:
                     if len(block_batch) < 32:
                         continue
 
+                    if profile:
+                        pr = cProfile.Profile()
+                        pr.enable()
+                        receive_start_time = time()
```

Comment:
I was not aware of `time.monotonic()`. There are a lot of places where we use the system clock to time things we should use a monotonic clock for.

Reply:
There's some nuance to switching to this in that we also should have a lot of stuff use a clock we can take control over.  For example when testing timeouts if you have a controllable clock then it can fastforward through the, say, 30 second timeout and your test can execute faster.  But, I haven't dug into the options for asyncio in detail yet and probably lots of code woul have to be tweaked.  Anyways, just a heads up that there may be another clock suggestion coming at some point.

Reply:
this is not a simulation though. This is profiling the CPU usage, and this clock needs to be real-time, not simulated time. But I agree that our simulations really would benefit from using simulated time, which would require some significant surgery in the existing code.

Reply:
Agreed that it controlled time is not applicable here.  I made the comment since using `time.monotonic()` applies lots of places I didn't want you surprised when in a few months I come back and say that it is also wrong in lots of those places because we are adding a controlled time source.  Anyways, the suggested change has been made.  Thanks.

Reply:
agreed. I'm planning a few PRs to switch out `time.time()` in a few places.